packages/ra-core/src/controller/ListController.tsx

```diff
@@ -194,25 +194,34 @@ export class UnconnectedListController extends Component<Props> {
         this.setFilters.cancel();
     }
 
-    componentWillReceiveProps(nextProps: Props) {
+    componentDidUpdate(prevProps: Props) {
+        const {
+            resource,
+            query,
+            params,
+            filter,
+            sort,
+            perPage,
+            version,
+        } = this.props;
+
         if (
-            nextProps.resource !== this.props.resource ||
-            nextProps.query.sort !== this.props.query.sort ||
-            nextProps.query.order !== this.props.query.order ||
-            nextProps.query.page !== this.props.query.page ||
-            nextProps.query.perPage !== this.props.query.perPage ||
-            !isEqual(nextProps.query.filter, this.props.query.filter) ||
-            !isEqual(nextProps.filter, this.props.filter) ||
-            !isEqual(nextProps.sort, this.props.sort) ||
-            !isEqual(nextProps.perPage, this.props.perPage)
+            prevProps.resource !== resource ||
+            prevProps.query.sort !== query.sort ||
+            prevProps.query.order !== query.order ||
+            prevProps.query.page !== query.page ||
+            prevProps.query.perPage !== query.perPage ||
+            !isEqual(prevProps.query.filter, query.filter) ||
+            !isEqual(prevProps.query.filter, query.filter) ||
+            !isEqual(prevProps.params, params) ||
```

Comment:
you've added the params key here, it has nothing to do with the bug you mentioned. Please remove.